### DISCUSSION TITLE
Should we use a single format when referring to bips in docs

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1097] [Q&A]
Warning: Quite trivial.

For devs with OCD it is a little bit annoying that we have various ways to refer to bips in our docs. Should/can we have a single preferred way so I can stop thinking about it each time I write it?

Currently we have (using 34 as an example)
- bip34
- Bip34
- BIP34
- BIP-34
- BIP 34

Just to be clear, I"m not suggesting enforcing this when reviewing, I'm just looking for rough consensus so I can personally favour one form and use it when cleaning up the codebase.

This sort of policy question comes up a bit for me, perhaps we could have a section in the README
```
### Soft Policies

We have various coding conventions that are specific to rust-bitcoin, these are not hard-and-fast rules and you should not worry about them too much if you are new to the project (or disagree). Others may 'clean up' your code after it has been merged to match these policies.

- Use the following form for referring to bips: <TBD>
- Put a line of whitespace after _all_ headings in rustdocs
- Be liberal with linking to relevant bips, it helps others learn. <insert example below>
- etc.
``` 

In case absolutely no on cares, I'll put forward `BIP-34` as the proposed preferred form.

**Comment by dpc** [comment]
> Just to be clear, I"m not suggesting enforcing this when reviewing,

Write a quick CI pass to enforce it instead. :)

**Reply by DanGould** [reply]
when I search for bip repo mediawikis I always search bip-{d:0>4} eg bip-0034 because that's how the file is named. When I search a codebase I search bip{d} e.g bip34 because that's how I've seen crates and variables named.  Therefore I'm in favor of case insensitive no separator. It's especially tough to find bip references in issues for the reasons you stated

**Reply by tcharding** [reply]
Ahh, keeping doc form and variable form the same - nice!

**Reply by DanGould** [reply]
hmmm but looking at bip specs (eg 174) I see them usually say "BIP 34" in the doc body

**Reply by Kixunil** [reply]
Perhaps put it into git hook but not CI? Seems too nitpicky to waste time on enforcing.

**Comment by Kixunil** [comment]
AFAIK BIP-42 is the canonical way humans write it in text. I myself don't bother with the hyphen because I'm lazy but I know I should use it. :)

**Comment by apoelstra** [comment]
I have no preference. I don't think I'm consistent when writing this. If we come up with a rule I"ll try to follow it.
